File: lab-java/src/main/java/br/com/ronistone/labapp/service/impl/CarServiceImpl.java
package br.com.ronistone.labapp.service.impl;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.com.ronistone.labapp.model.Car;
import br.com.ronistone.labapp.repository.CarRepository;
import br.com.ronistone.labapp.service.CarService;

@Service
public class CarServiceImpl implements CarService {

    @Autowired
    private CarRepository carRepository;


    @Override
    public Optional<List<Car>> getAllCars() {
        return Optional.of(carRepository.findAll());
    }

    @Override
    public Optional<Car> getCarById(Long id) {
        return carRepository.findByIdWithSleep(id);
    }

}

File: lab-kotlin/src/main/kotlin/br/com/ronistone/labkotlin/service/impl/CarServiceImpl.kt
package br.com.ronistone.labkotlin.service.impl

import br.com.ronistone.labkotlin.model.Car
import br.com.ronistone.labkotlin.repository.CarRepository
import br.com.ronistone.labkotlin.service.CarService
import org.springframework.stereotype.Service

@Service
class CarServiceImpl(
        val carRepository: CarRepository
): CarService {

    override fun getAllCars(): MutableList<Car>? {
        return carRepository.findAll();
    }

    override fun getCarById(id: Long): Car? {
        return carRepository.findByIdWithSleep(id)
    }

}
